File: ruby/charges_paginate.rb
require 'stripe'

Stripe.api_key = "sk_your_api_key"

charges = Stripe::Charge.all(:limit => 100)

charges.each do |charge|
  puts charge.id
end

while charges.has_more do 
  charges = Stripe::Charge.all(:limit => 100, :starting_after => charges.data.last.id)

  charges.each do |charge|
    puts charge.id
  end
end

File: python/charges_paginate.py
import stripe
stripe.api_key = "sk_your_api_key"

charges = stripe.Charge.all(limit=100)

for charge in charges.data:
  print(charge.id)

while charges.has_more:
  charges = stripe.Charge.all(limit=100, starting_after=charges.data[-1])
  for charge in charges.data:
    print(charge.id)